pkg/virt-launcher/virtwrap/libvirtxml/convert.go

```diff
@@ -300,6 +300,7 @@ func ConvertKubeVirtFeaturesToDomainFeatureList(features *api.Features) *libvirt
 	f.PMU = setDomainFeatureState(features.PMU)
 	f.HyperV = ConvertKubeVirtFeatureHypervToDomainFeatureHyperV(features.Hyperv)
 	f.KVM = ConverKubeVirtFeatureKVMToDomainFeatureKVM(features.KVM)
+	f.VMPort = setDomainFeatureState(features.VMPort)
```

Comment:
@vladikr IMHO this looks like way to self sabotage...  a new feature would break us...

Reply:
Can you elaborate please?

Reply:
@xpivarc I agree. I dislike this `ConvertKubeVirtFeaturesToDomainFeatureList` 
Do you have a suggestion?

Reply:
Why do we set this again instead of simply taking these from the existing domain XML?

Reply:
+1
The whole logic under `migratableDomXML`'s `if vmi.IsCPUDedicated()` was very confusing and surprising to me. This also caused the failures mentioned [above](https://github.com/kubevirt/kubevirt/pull/13252#issuecomment-2502006341).

https://github.com/kubevirt/kubevirt/blob/e2e64962d9db7dc3724f751eb6c897215b5842a2/pkg/virt-launcher/virtwrap/live-migration-source.go#L212-L226

Do you understand why we need this logic in the first place? I'd be happy to refactor in a follow up.

Reply:
We need to unmarshal the domain into the libvirtxml representation because right now unmarshaling to our representation is lossy.  I am not sure if this was the case before migrating to libvirtxml or not...

Reply:
@iholder101 @vladikr We should also tests that the function is lossless